Search - 'Clear all Filters' doesn't work

YuliaKrimerman opened the pull request:
<!--- If this is a non-code change, this template is not required; reference any issues or top-level descriptions as needed -->
<!--- All code change PRs should relate to an issue, reference it here; see example below -->
<!--- https://issues.redhat.com/browse/RHOAIENG-12861 -->
https://issues.redhat.com/browse/RHOAIENG-12861

## Description
<!--- Describe your changes in detail; the what, the why, any findings, etc -->
<!--- Include any screenshots of changed UI; Include any gifs if it was a flow / UX change -->
1. Clear all filters button is now working on all existing screens 
2.  Changed label that was mentioned in RHOAI sprint demos.

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
on the UI

## Test Impact
<!--- What tests have you done to covert implemented functionality -->
<!--- If tests are not applicable, explain why here -->
Updated tests to pass after the label change. In order to test the change, verify that now "Clear all Filters' button is now working everywhere it exists 

## Request review criteria:
<!--- This PR will be merged by any repository approver when it meets all the points in the checklist -->
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 
<!--- You can ignore these if you are doing manifest, backend, internal logic, etc changes; aka non-UI / visual changes -->
- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

ppadti commented:
@YuliaKrimerman I think we need to fix this issue in Deployed models page as well.

YuliaKrimerman commented:
@ppadti I was trying to add that fix into InferenceServiceTable inside the Table, but even though it has emptyTableView that whole chip group is not showing, what am I missing?

manaswinidas reviewed (commented):
@ppadti I don't think we have a toolbar with filters for the deployments tab here as per the [mocks](https://www.figma.com/proto/KTsYhANYQqB0NUrXW7wQH1/Model-Registry?page-id=6782%3A296930&node-id=8238-700071&viewport=1846%2C28%2C0.43&t=anh0eGvvgwFUHiMb-1&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=8238%3A699060&show-proto-sidebar=1). Let me know if you're referring to some other mock links.

manaswinidas commented on frontend/src/pages/modelRegistry/screens/RegisteredModels/RegisteredModelListView.tsx:
```diff
@@ -105,12 +105,19 @@ const RegisteredModelListView: React.FC<RegisteredModelListViewProps> = ({
     </ToolbarGroup>
   );
 
+  const clearAllFilters = resetFilters;
```
I don't think there is a need to create a new variable when we can just use `resetFilters` everywhere. Also, I see `() => setSearch('')` present in the same file numerous times when `resetFilters` function is doing the same thing. Can we replace all of them with `resetFilters`? Otherwise, we can rename `resetFilters` to `clearAllFilters` but all `() => setSearch('')` should be replaced with that function.

ppadti commented:
> @ppadti I don't think we have a toolbar with filters for the deployments tab here as per the [mocks](https://www.figma.com/proto/KTsYhANYQqB0NUrXW7wQH1/Model-Registry?page-id=6782%3A296930&node-id=8238-700071&viewport=1846%2C28%2C0.43&t=anh0eGvvgwFUHiMb-1&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=8238%3A699060&show-proto-sidebar=1). Let me know if you're referring to some other mock links.

Yeah, I was referring to this [slack conversation](https://redhat-internal.slack.com/archives/C07C0447EAV/p1727361566973249) here.

YuliaKrimerman commented:
@ppadti , @manaswinidas  As I mentioned above I was trying to add that fix into InferenceServiceTable(it's the right file correct?) inside the Table, but even though it has emptyTableView that whole chip group is not showing, what am I missing? Any suggestions? And yes, I suggested to fix it as part of that PR in that conversation, but just got stuck, and preferred to, at least put the required solution for the actual bug

ppadti commented:
> @ppadti , @manaswinidas As I mentioned above I was trying to add that fix into InferenceServiceTable(it's the right file correct?) inside the Table, but even though it has emptyTableView that whole chip group is not showing, what am I missing? Any suggestions? And yes, I suggested to fix it as part of that PR in that conversation, but just got stuck, and preferred to, at least put the required solution for the actual bug

May be because it is using DashboardSearchField and not ToolbarFilter and we have this issue in some other places as well for ex: Accelerator profile and Notebook images page. I think we need to discuss this with the team and can have a separate issue to address this. wdyt?

YuliaKrimerman commented:
Sounds good to me, I can open an issue for it, and will address the related PR comments then.

YuliaKrimerman commented:
@manaswinidas Addressed comment, ready for re-review. Also created a story for us to discuss https://issues.redhat.com/browse/RHOAIENG-13924 . @ppadti . Thank you!

ppadti commented:
tested locally, clear all filters works well.
/lgtm

mturley reviewed (approved):
LGTM, thanks @YuliaKrimerman
